jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/ResourceUtil.java

```diff
@@ -50,6 +55,42 @@ public static boolean jsonEquals(JsonObject first, JsonObject second) {
         return getObjectMapper(ResourceFileType.fromFile(file)).readerFor(clazz).<T>readValues(file).readAll();
     }
 
+    public static List<KubernetesResource> loadKubernetesResourceList(File file) throws IOException {
+        ResourceFileType resourceFileType = ResourceFileType.fromFile(file);
+        List<KubernetesResource> kubernetesResources = new ArrayList<>();
+        if (file.isFile()) {
+            String fileContentAsStr = new String(Files.readAllBytes(file.toPath()));
+            if (StringUtils.isNotBlank(fileContentAsStr)) {
+                kubernetesResources.addAll(loadKubernetesResourceListFromString(resourceFileType, fileContentAsStr));
+
+            }
+        }
+        return kubernetesResources;
+    }
+
+    private static List<KubernetesResource> loadKubernetesResourceListFromString(ResourceFileType resourceFileType, String fileContentAsStr) throws JsonProcessingException {
+        Map<String, Object> listAsMap = getObjectMapper(resourceFileType).readValue(fileContentAsStr, Map.class);
+        List<KubernetesResource> kubernetesResources = new ArrayList<>();
+        List<Map<String, Object>> items = (List<Map<String, Object>>)listAsMap.get("items");
+        for (Map<String, Object> item : items) {
+            KubernetesResource resource = getHasMetadataOrGenericResource(resourceFileType, item);
+            kubernetesResources.add(resource);
+        }
+        return kubernetesResources;
+    }
+
+    private static KubernetesResource getHasMetadataOrGenericResource(ResourceFileType resourceFileType, Map<String, Object> item) {
+        try {
+            return convertValue(resourceFileType, item, KubernetesResource.class);
+        } catch (IllegalArgumentException illegalArgumentException) {
+            return convertValue(resourceFileType, item, GenericCustomResource.class);
```

Comment:
When is this going to happen?
`GenericCustomResource` implements `KubernetesResource`, I don't understand what we're covering here.

Reply:
OK, so after running the tests, the question is just the opposite, why is this method not just (or even removed)?:
```
    private static GenericCustomResource getHasMetadataOrGenericResource(ResourceFileType resourceFileType, Map<String, Object> item) {
        return convertValue(resourceFileType, item, GenericCustomResource.class);
    }
```

Reply:
OK, I see the point now, that when deserializing to `KubernetesResource.class` we will make use of the KubernetesDeserializer, as opposed to what happens when using the `GenericCustomResource.class` as target, since this one overrides the JsonDeserialize annotation.

However, this is really hidden and the code doesn't really help to understand what's going on.